### DISCUSSION TITLE
Is awaiting on a mutation required?

### DISCUSSION THREAD

**Discussion by louisrli** [prisma/prisma#3781] [Q&A] [answered]
If I have an express server graphql-yoga, and I do something like:

```ts
const myResolver = async (parent, args, ctx) => {
    db.myEntity.update(...)
    return "Some resolver response";
}
```

is this guaranteed to fire the update regardless of whether I await on it or not? I vaguely remember in Prisma 1 that you had to await on something in order for the Prisma action to actually execute but I'm not sure if I'm misremembering.

**Comment by ryands17** [comment, marked as answer]
Hey @louisrli  👋 

Yes you need await because the calls made to Prisma client are lazy. They are only evaluated when `await` is called, which is why the first one doesn't work. You can also read more about this [here](https://github.com/prisma/docs/issues/800).

**Reply by louisrli** [reply]
Thanks for the answer. Just wanted to check, this makes sure it's not possible to optimize certain flows, for example, this would not be possible: "I want to fire and forget this update but I don't want it to block the return of the request." Is that correct?

**Reply by ryands17** [reply]
You could use [transactions](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/transactions#overview) for that specific use case. They are not lazy afaik.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
